### ISSUE TITLE
Can this dataset be registered?

### ISSUE THREAD

**Issue by zzx332** [voxelmorph/voxelmorph#184]
HI,
I want to do multimodal registration with MI in CHAOS(a liver dataset with 40ct,40mri).But the number of MRI slices is very small(Average about 30.)Can it work（if interpolate to 128）？
<img width="640" alt="微信截图_20200504212939" src="https://user-images.githubusercontent.com/45729525/80970854-69181280-8e4e-11ea-953f-418ea7c81582.png">
I try to use your loss Global MI in other registration network to do CT-CT registration in image of liver.
since the value obtained by mi is too small(Initially only 0.03,Is this normal？),I set 
MI_loss=10*globalmi,loss=MI_loss+smooth_loss.
The following is the loss training results(raw_loss is MI_loss,reg_loss is smooth_loss), it seems that the test results are overfitting.
<img width="260" alt="微信截图_20200504215825" src="https://user-images.githubusercontent.com/45729525/80973814-71724c80-8e52-11ea-9f9e-148edaa723c0.png">
<img width="260" alt="微信截图_20200504215835" src="https://user-images.githubusercontent.com/45729525/80973802-6f0ff280-8e52-11ea-950c-4c3e6beb2eaf.png">

**Comment by adalca** [comment]
Hi @zzx332 

Sorry, there are a whole slew of questions there

Can you register MR to CT? Sure, I would encourage you to look at https://arxiv.org/abs/2004.10282 and try our recent code in the redesign branch, perhaps trying an affine registration first.

Can you register CT to CT images of liver? Sure, for that I don't think you need NMI (NCC should work since they are both CT). I'm not quite sure why you think that the fields are overfitting. Did you try visualizing results?

**Comment by zzx332** [comment]
Here is my result on MR to CT.the first is mi_loss，mi_loss=-10*globalmi(If set to mi_loss=-globalmi,miloss does not decrease).The second one is smooth_loss.Loss = mi_loss+smooth_loss

Mi_loss:https://user-images.githubusercontent.com/45729525/81144615-08e8b400-8fa7-11ea-8c3e-e33d36cf8190.png
smooth_loss:https://user-images.githubusercontent.com/45729525/81144674-2584ec00-8fa7-11ea-9b95-d844c5842598.png
It looks like globalloss is not converging,smooth_loss first decreases and then increases.
Here is dice on the test set.It looks ok, but the warped image is bad.
val_dice:https://user-images.githubusercontent.com/45729525/81145967-f58b1800-8fa9-11ea-88c4-2643784840e6.png
img_fixed:https://user-images.githubusercontent.com/45729525/81146587-2ddf2600-8fab-11ea-9aca-071f6256059e.jpg
img_moving:https://user-images.githubusercontent.com/45729525/81146583-2a4b9f00-8fab-11ea-962c-4ec433ff3cd6.jpg
img_warped:https://user-images.githubusercontent.com/45729525/81146166-531f6480-8faa-11ea-8130-bada6792eb4f.jpg
Is there any problem with my loss function settings?

**Comment by adalca** [comment]
It looks to me like you have not run the algorithm for long enough, as it is still converging.

Although we haven't played with the MI loss too much, setting the mi_loss to 10 is too larger I think, you should definitely try to make it work with 1. I don't know what you mean by 'does not decrease' -- can you show a screenshot? If it decreases with 10 I'd be very surprised if it doesn't decrease with 1.

**Comment by zzx332** [comment]
Thank you for your kindful help！
I run the algorithm for long enough,but it seems that mi_loss has not fully converged:
https://user-images.githubusercontent.com/45729525/81475749-a6124980-9240-11ea-9c7f-11cca63048c5.png
At the same time my smoothloss has been rising：
https://user-images.githubusercontent.com/45729525/81475774-dce85f80-9240-11ea-990c-974230d8a0b1.png
Loss=-mi_loss+0.1smoothloss
Is this due to the big difference between my data（ct MRI liver）？
MRI：https://user-images.githubusercontent.com/45729525/81475842-2cc72680-9241-11ea-9aab-4dbf7a8d8da5.png
CT：https://user-images.githubusercontent.com/45729525/81475854-39e41580-9241-11ea-8f24-2b6705f2f657.png
I read your paper on multimodality:https://arxiv.org/abs/2004.10282
Very good paper.The article mentions that the code is in the branch,but I did not find the code implementation from the branch.
Can you share the T2 atlas ？I don't know how to build a qualified T2 altas with segmentation.
my graduation project does not have enough data...

**Comment by adalca** [comment]
Hi @zzx332 

Sorry for the delay. We've been working hard on the new version of the vxm library. 
Could you check if the new version, in the [`master` branch](https://github.com/voxelmorph/voxelmorph/tree/master) leads to the same issues?
